Observer/IBotObserver.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WeatherMonitoringSystem.model;
+
+namespace WeatherMonitoringSystem.Observer
+{
+    public interface IBotObserver
```

Comment:
Maybe `IWeatherObserver` instead? we should name the observer according to what it should be observing. The main reason for using an interface is to abstract the actual implementation, so we don't care if it is a bot or any kind of class, all we care about is that it's _observing_ the _weather_.